dom.bs

```diff
@@ -6958,8 +6960,31 @@ a boolean <var>delegatesFocus</var>, and a boolean <var>slotAssignment</var>:
   <p>If <var>element</var> is a <a for=Element>shadow host</a>, then:
 
   <ol>
-   <li><p>If <var>element</var>'s <a for=Element>shadow root</a>'s <a for=ShadowRoot>declarative</a>
-   is false, then <a>throw</a> an "{{NotSupportedError!!exception}}" {{DOMException}}.
+   <li><p>Let <var>currentShadowRoot</var> be <var>element</var>'s
+   <a for=Element>shadow root</a>.
+
+   <li><p>If one of the following is true:
+
+   <ul>
+    <li><p><var>currentShadowRoot</var>'s <a for=ShadowRoot>declarative</a> is false,
+
+    <li><p><var>currentShadowRoot</var>'s <a for=ShadowRoot>mode</a> does not match
+    <var>mode</var>,
+
+    <li><p><var>currentShadowRoot</var>'s <a for=ShadowRoot>delegates focus</a> does
+    not match <var>delegatesFocus</var>,
+
+    <li><p><var>currentShadowRoot</var>'s <a for=ShadowRoot>slot assignment</a> does
+    not match <var>slotAssignment</var>,
+
+    <li><p><var>currentShadowRoot</var>'s <a for=ShadowRoot>clonable</a> does
+    not match <var>clonable</var>,
+   </ul>
+
+   <p>then <a>throw</a> a "{{NotSupportedError!!exception}}" {{DOMException}}.
+
+   <p class=note>Any new parameters added to <a>attach a shadow root</a> should be
```

Comment:
Notes cannot contain "should". Maybe this should just be a source code comment directly under the start of the "attach a shadow root" algorithm? It's something we shouldn't forget about after all.

Reply:
Done. LMK if you like the wording.